Tests/ParserTests/type_states.flint

```diff
@@ -0,0 +1,16 @@
+// RUN: %flintc %s --dump-ast | %FileCheck %s --prefix CHECK-AST
+
+contract CapabilityBinding {}
+
+// CHECK-AST: ContractBehaviorDeclaration
+// CHECK-AST:   identifier "CapabilityBinding"
+// CHECK-AST:   TypeState
+// CHECK-AST:     identifier "stated"
+// CHECK-AST:   CallerCapability
+// CHECK-AST:     identifier "any"
+CapabilityBinding @(stated) :: (any) {
```

Comment:
Space? Not actually against this syntax but was wondering why you did it this way

Reply:
I did it this way to simplify the parsing slightly. Shall we merge it how it is, and then change it later if required.